barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover_impl.hpp

```diff
@@ -55,8 +55,9 @@ ProtogalaxyProver_<DeciderProvingKeys>::perturbator_round(
     const FF delta = transcript->template get_challenge<FF>("delta");
     const std::vector<FF> deltas = compute_round_challenge_pows(CONST_PG_LOG_N, delta);
     // An honest prover with valid initial key computes that the perturbator is 0 in the first round
-    const Polynomial<FF> perturbator = accumulator->is_accumulator ? Fun::compute_perturbator(accumulator, deltas)
-                                                                   : Polynomial<FF>(CONST_PG_LOG_N + 1);
+    const Polynomial<FF> perturbator = accumulator->is_accumulator
+                                           ? Fun::compute_perturbator(accumulator, deltas, trace_usage_tracker)
```

Comment:
Can you not pass this usage tracker as a function argument here or anywhere else? I worked hard to make these things pure functions to highlight what is going on in the protocol and I think it's better to put this extra structure in the background as mutable class data.